### PR TITLE
Update map_tree.py

### PR DESCRIPTION
**Pull request by hayalaso**
Moving the `roimap` array as a table instead of being in the header when using `HealpixMapROI`.

The current patch can write hdf5 files now. Reading the HDF5 using the script `hal_hdf5_to_fits.py` gives a bit of problems since hal expects to read the `roimap` from the header. I'll see if this can be modified somehow.

### CODE REVIEW

**Comment by hayalaso** [comment]
I tested the changes using both `HealpixROIMap` and `HealpixConeMap`. I also used the script `hal_hdf5_to_fits.py` and it seems it works for both of them. I also noticed, although probably not that important, that `tables 3.5.2` or bigger was needed. Maybe not necessary in py3.

**Comment by henrikef** [comment]
@hayalaso  is this ready for review or are you still working on it?

Could you add something to the tests as well?

**Comment by hayalaso** [comment]
I got busy with other things the past couple of weeks so I  don't remember. I'll need to double-check.

**Comment by henrikef** [comment]
Btw, on github, using `WIP` in the pull request name doesn't do anything. Instead, you can mark pull requests as draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/ You can also convert to draft later, there's a link on the upper right, underneath where you can select reviewers.

**Comment by hayalaso** [comment]
I added a test to write a model map using the HealpixROIMap and also read it back. I think it should be ready to merge.

File: tests/test_healpixRoi.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+from conftest import check_map_trees
+
+
+def Sky2Vec( ra, dec ):
+  c = SkyCoord( frame = "icrs", ra=ra*u.degree, dec=dec*u.degree )
+  theta = (90.0*u.degree-c.dec).to(u.radian).value
+  phi = c.ra.to(u.radian).value
+  vec = hp.pixelfunc.ang2vec(theta, phi)
+  return vec
```
Comment (henrikef) [inline tests/test_healpixRoi.py]: Just a suggestion, but woulnd't 

```
def Sky2Vec(ra, dec):
    return hp.pixelfunc.ang2vec(ra, dec, lonlat=True )
```
 do the same thing?

**Review by henrikef — APPROVED** [review]
Looks good to me. Once the tests pass, this should be ready to merge.

**Comment by henrikef** [comment]
Hi @hayalaso!

Thank you for implementing this!

Please pull in the most recent version of the hawc_hal master into your branch and resolve any conflicts. Note that the tests directory moved to hawc_hal/tests.

Once the tests pass, this is ready to merge.

**Comment by henrikef** [comment]
Fixes #22 

Fixes #55 

Hm. This isn't working for some reason. Maybe the author has to do it? Might need to close those issues manually.

File: tests/test_healpixRoi.py (hunk 1 above)
Comment (hayalaso) [reply tests/test_healpixRoi.py]: Mmm, probably. I copied that function from another test.

**Pull request merged**